tools/vim/.ycm_extra_conf.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2016 Marcin Copik
+# Copyright (c) 2016 Parsa Amini
+#
+#  Distributed under the Boost Software License, Version 1.0. (See accompanying
+#  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+import os
+import re
+
+compilation_database_dir = "@CMAKE_BINARY_DIR@"
+
+def FlagsForFile(filename, **kwargs):
+
+    a = ''
+    #Ignore lines not containing CXX command
+    with open('{0}/compile_commands.json'.format(compilation_database_dir), 'r') as f:
```

Comment:
instead of string operations, wouldn't path operations be a bit more appropriate?